ATM.java

```diff
@@ -0,0 +1,136 @@
+                    int acc = in.nextInt();
+                    System.out.print("How many credits do you want to withdraw? ");
+                    int amount = in.nextInt();
+                    // withdraw from checkings or savings
+                    if (acc == 1) {
+                        // withdraw from checkings
+                        int newTotal = checking.withdrawMoney(amount);
+                        if (newTotal < 0) {
+                           System.out.println("Insufficient funds. ");
+                        }
+                        System.out.println("Your savings account has " + savings.getSavingsBalance() + " credits. ");
+                        System.out.println("Your checking account has " + checking.withdrawMoney(amount) + " credits. \n");
+                    }
+                    else {
+                        // withdraw from savings
+                        double newTotal = savings.withdrawMoney(amount);
+                        if (newTotal < 0) {
+                           System.out.println("Insufficient funds. ");
+                        }
+                        System.out.println("Your savings account has " + savings.withdrawMoney(amount) + " credits. ");
+                        System.out.println("Your checking account has " + checking.getCheckingBalance() + " credits. \n");
+                    }
+                    break;
+                }
+                case "C":
+                case "c":
+                    System.out.println("Your savings account has " + savings.getSavingsBalance() + " credits. ");
+                    System.out.println("Your checking account has " + checking.getCheckingBalance() + " credits. \n");
+                    break;
+                // Transfering
+                case "T":
+                case "t": {
+                    System.out.print("Transfer from: (1)Checking to Savings or (2)Savings to Checking? ");
+                    int acc = in.nextInt();
+                    System.out.print("Amount to transfer? ");
+                    int amount = in.nextInt();
+                    // transfer from checkings or savings
+                    if (acc == 1) {
+                        // deposit amount into savings
+                        System.out.println("Your savings account has " + savings.depositMoney(amount) + " credits. ");
+                        // withdraw amount from checking
+                        System.out.println("Your checking account has " + checking.withdrawMoney(amount) + " credits. \n");
+                    }
+                    else {
+                        // withdraw amount from savings
+                        System.out.println("Your savings account has " + savings.withdrawMoney(amount) + " credits. ");
+                        // deposit amount into checking
+                        System.out.println("Your savings account has " + checking.depositMoney(amount) + " credits. ");
```

Comment:
The problem here is if I withdraw more money than I have in my savings account, it will still deposit that amount in my checking account. Same with the transaction above.